src/doctopus/storage.clj

```diff
@@ -0,0 +1,46 @@
+(ns doctopus.storage
+  "Defines how Doctopus speaks to different places it can store generated HTML"
+  (:require [doctopus.storage-impls :as storage-impls]))
+
+;; ## Storage
+;; Doctopus shouldn't really care _where_ it puts generated HTML; it
+;; just needs a consistent interface. Thus, the Backend: a way of
+;; generalizing and controlling where Doctopus tries to get and put Stuff.
+;;
+;; As a first pass, the backend will be defined by an atom which
+;; contains the current Backend instance; functions in this NS will
+;; work against that atom. This means a Doctopus will have a single Backend; we
+;; can potentially expand this later.
+;;
+;; Initial backends:
+;; - Temporary filesystem (provides it's own directory)
+;; - Permanent filesystem (needs a directory provided)
+;; - Amazon S3 (maybe?)
+;;
+;; At time of writing, I'm thinking of loading and saving one Tentacle worth of
+;; documents all at once.
+
+(defprotocol DoctopusBackend
+  (set-backend! [this backend])
+  (load-from-storage [this k])
+  (save-to-storage [this k v]))
```

Comment:
Depends on how the overwrites work. You should probably generate HTML into
a directory, then move it over top of the existing one.

On Wed, Apr 22, 2015 at 1:49 PM, jesse keane <notifications@github.com>
wrote:

> In src/doctopus/storage.clj
> <https://github.com/Gastove/doctopus/pull/6#discussion_r28913294>:
>
> > +;; contains the current Backend instance; functions in this NS will
> > +;; work against that atom. This means a Doctopus will have a single Backend; we
> > +;; can potentially expand this later.
> > +;;
> > +;; Initial backends:
> > +;; - Temporary filesystem (provides it's own directory)
> > +;; - Permanent filesystem (needs a directory provided)
> > +;; - Amazon S3 (maybe?)
> > +;;
> > +;; At time of writing, I'm thinking of loading and saving one Tentacle worth of
> > +;; documents all at once.
> > +
> > +(defprotocol DoctopusBackend
> > +  (set-backend! [this backend])
> > +  (load-from-storage [this k])
> > +  (save-to-storage [this k v]))
>
> any need to ever remove-from-storage?
> i can't really think of one, but eh?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Gastove/doctopus/pull/6/files#r28913294>.
>